### PR TITLE
Added mode selection and command parser library

### PR DESCRIPTION
**Pull request by liealex**

### CODE REVIEW

File: examples/first-frame/CMakeLists.txt [hunk 1]
```diff
@@ -28,3 +28,5 @@ if( WIN32 )
         )
     endif()
 endif()
+
+target_link_libraries(${PROJECT_NAME} PRIVATE aditof command_parser)
```
Comment (dNechita) [inline examples/first-frame/CMakeLists.txt:32]: first-frame already links to "aditof" (see line 8). You should add command_parser there as well.

Comment (liealex) [reply examples/first-frame/CMakeLists.txt]: Solved

File: examples/first-frame/main.cpp [hunk 2]
```diff
@@ -41,9 +42,35 @@
...
 using namespace aditof;
 
+static const char Help_Menu[] =
+    R"(First-frame usage:
+    first-frame [-config | --CONFIG <CONFIG>] 
+    first-frame (-h | --help)
+    first-frame [-ip | --ip <ip>] [-m | --m <mode>] [--config | --CONFIG <CONFIG>]
+
+    Arguments:
+      [-config | --CONFIG <CONFIG>]            Input config_default.json file (which has *.ccb and *.cfg)
```
Comment (rbudai98) [inline examples/first-frame/main.cpp:56]: `-config` is also mismatch from the previous line, `--config`

Comment (liealex) [reply examples/first-frame/main.cpp:56]: Typo's for mandatory argument <CONFIG>in the help menu will be solved once the parser update will be merged (mandatory arguments will no longer need a prefix of type "-"/"--" so they will be deleted)

File: examples/first-frame/main.cpp [hunk 3]
```diff
@@ -71,33 +98,96 @@ Status save_frame(aditof::Frame &frame, std::string frameType) {
 }
 
 int main(int argc, char *argv[]) {
+    std::map<std::string, struct Argument> command_map = {
+        {"-h", {"--help", false, "", ""}},
+        {"-ip", {"--ip", false, "", ""}},
+        {"-m", {"--m", false, "", "0"}},
+        {"-config", {"-CONFIG", true, "last", ""}}};
```
[the remark this answers is not in the document]
Comment (liealex) [reply examples/first-frame/main.cpp:105]: Solved

File: examples/first-frame/main.cpp [hunk 4]
```diff
@@ -71,33 +98,96 @@ Status save_frame(aditof::Frame &frame, std::string frameType) {
...
+        {"-m", {"--m", false, "", "0"}},
+        {"-config", {"--CONFIG", true, "last", ""}}};
 
+    CommandParser command;
+    std::string arg_error;
     google::InitGoogleLogging(argv[0]);
     FLAGS_alsologtostderr = 1;
 
+    command.parseArguments(argc, argv);
+    int result = command.checkArgumentExist(command_map, arg_error);
```
[the remark this answers is not in the document]
Comment (liealex) [reply examples/first-frame/main.cpp:113]: Solved

File: examples/first-frame/CMakeLists.txt [hunk 5]
```diff
@@ -27,4 +27,4 @@ if( WIN32 )
         COMMAND ${CMAKE_COMMAND} -E copy_directory ${GLOG_BIN_DIR} ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>
         )
     endif()
-endif()
+endif()
```
Comment (rbudai98) [inline examples/first-frame/CMakeLists.txt:30]: new line missing

File: examples/first-frame/main.cpp [hunk 6]
```diff
@@ -41,9 +42,35 @@
...
+
+    Valid mode (-m | --m) options are:
+        0: short-range native
+        1: long-range native
+        2: short-range Qnative
+        3: long-range Qnative
+        4: pcm-native
+        5: long-range mixed
+        6: short-range mixed
+    )";
```
Comment (rbudai98) [inline examples/first-frame/main.cpp:72]: Might want to remove any extra space  before `)`, or perhas adding an extra new line. This way the new command line wont be disalligned

Comment (liealex) [reply examples/first-frame/main.cpp]: Solved

File: examples/first-frame/CMakeLists.txt (hunk 5 above)
Comment (liealex) [reply examples/first-frame/CMakeLists.txt]: Solved

**Pull request merged**